include/range/v3/utility/associated_types.hpp

```diff
@@ -29,52 +29,41 @@ namespace ranges
         /// @{
 
         ////////////////////////////////////////////////////////////////////////////////////////
+        /// \cond
+        namespace detail
+        {
+            template<typename T, typename Enable = void>
+            struct difference_type2
+            {};
+
+            template<typename T>
+            struct difference_type2<T, meta::void_<decltype(std::declval<const T>() - std::declval<const T>())>>
```

Comment:
So, just because something can be subtracted, `difference_type` should work for that type? That is not what we have currently specified. And `make_signed` is unlikely to so anything sane with a type that is not `Integral`.